src/Behat/Context/EntityContext.php

```diff
@@ -314,44 +314,95 @@ protected function checkEntityValues($entity, $fields) {
     return $errors;
   }
 
+  /**
+   * Make replacement on given string when this one contains entity-replacement.
+   *
+   * [entity-replacement:{entity_type}:{field_name}:{field_value}:{destiny_replacement_field_name}]
+   *
+   * entity_type -> the entity type on which to search.
+   * field_name -> the entity field to search on.
+   * field_value -> the value of the field that the entity should have.
+   * destiny_replacement_field_name -> The name of the field on which to take the value to be used as a replacement.
+   *
+   * Examples:
+   *  - [entity-replacement:user:mail:behat@metadrop.net:uid]
+   *    Output: 123
+   *  - [entity-replacement:user:mail:behat@metadrop.net:uid], [entity-replacement:user:mail:behat_2@metadrop.net:uid], [entity-replacement:user:mail:behat_3@metadrop.net:uid]
+   *    Output: 123, 124, 125
+   *  - [entity-replacement:user:mail:behat@metadrop.net:uid] - [entity-replacement:user:mail:behat_2@metadrop.net:uid] - [entity-replacement:user:mail:behat_3@metadrop.net:uid]
+   *    Output: 123 - 124 - 125
+   *  - entity-replacement:user:mail:behat@metadrop.net:uid (This way can only be use for single replacement)
+   *    Output: 123
+   *
+   * Every token will be replaced, and the text between them will
+   * remain the same.
+   *
+   * @param string $value
+   *   String that contains a single or multiple entity-replacement tokens.
+   *
+   * @return string|bool
+   *   Replaced or untouched string or FALSE otherwise.
+   *
+   * @throws \Exception
+   */
+  protected function entityTokensReplacements($value) {
+    $entity_tokens = [];
+    $replacements = [];
+    if (strpos($value, 'entity-replacement') === 0) {
+      $entity_tokens = (array) $value;
+    }
+    else {
+      $matches = [];
+      preg_match_all('#\[entity-replacement:?([^]]*)\]#', $value, $matches);
+      $entity_tokens = reset($matches);
+    }
+
+    if (empty($entity_tokens)) {
+      return FALSE;
+    }
+
+    // Get entity type list.
+    $entity_types = $this->getCore()->getEntityTypes();
+
+    foreach ($entity_tokens as $entity_token) {
+      $token_pieces = explode(':', str_replace(['[', ']'], ['', ''], $entity_token));
+      array_shift($token_pieces);
+      list($entity_type, $field_key, $field_value, $destiny_replacement) = $token_pieces;
+
+      if (!in_array($entity_type, $entity_types)) {
+        throw new \Exception(sprintf('The "%s" token or its entity values are not valid!', $entity_token));
+      }
+
```

Comment:
This is needed to prevent warnings when using the list method:

```suggestion
      if (count($token_pieces) < 4) {
         throw new \Exception(sprintf('Missing arguments to find the entity token with name: %s', $entity_token));
      }

      list($entity_type, $field_key, $field_value, $destiny_replacement) = $token_pieces;

      if (!in_array($entity_type, $entity_types)) {
        throw new \Exception(sprintf('The "%s" token or its entity values are not valid!', $entity_token));
      }

```

Reply:
Fixed